src/burnchains/burnchain.rs

```diff
@@ -777,24 +778,34 @@ impl Burnchain {
             })
             .unwrap();
 
+        let first_block_hash = self.first_block_hash.clone();
         let db_thread: thread::JoinHandle<Result<BurnchainBlockHeader, burnchain_error>> =
             thread::Builder::new()
                 .name("burnchain-db".to_string())
                 .spawn(move || {
                     let mut last_processed = burn_chain_tip;
                     while let Ok(Some(burnchain_block)) = db_recv.recv() {
-                        debug!("Try recv next parsed block");
+                        debug!("Try recv next parsed block, {:?}", &burnchain_block);
 
                         let block_height = burnchain_block.block_height();
 
                         let insert_start = get_epoch_time_ms();
-                        last_processed =
-                            Burnchain::process_block(&myself, &mut burnchain_db, &burnchain_block)
-                                .map_err(|e| {
-                                    warn!("Error processing block {:?}", e);
-                                    e
-                                })
-                                .unwrap();
+                        // The first block is added to Burnchain as a special case.
+                        if burnchain_block.block_hash() != first_block_hash {
+                            last_processed = Burnchain::process_block(
+                                &myself,
+                                &mut burnchain_db,
+                                &burnchain_block,
+                            )
+                            .map_err(|e| {
+                                warn!(
+                                    "Error processing block {:?}, error: {:?}, first_hash {:?}",
+                                    &burnchain_block, e, &first_block_hash
+                                );
```

Comment:
Logs like these should use the k-v syntax:

```suggestion
                                warn!(
                                    "Error processing block";
                                    "burnchain_block" => ?burnchain_block,
                                    "error" => ?e,
                                    "first_hash" => ?first_block_hash
                                );
```